### ISSUE TITLE
Enabling NixOS autoUpgrade doesn't work

### ISSUE THREAD

**Issue by AstroSnail** [NixOS/nixpkgs#100261]
**Describe the bug**
`system.autoUpgrade.enable = true;` creates a broken systemd service, which does not run successfully.

**To Reproduce**
1. In `configuration.nix` set `system.autoUpgrade.enable = true;`.
2. Switch/test this configuration.
3. Run `systemctl start nixos-upgrade.service`.
4. Observe it fail. It complains about an unrecognized flag `--no-build-output`.

**Expected behavior**
The service should run successfully, enabling the system to stay up-to-date automatically.

**Screenshots**
~~If applicable, add screenshots to help explain your problem.~~

**Additional context**
I think the problematic code is here:
https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/tasks/auto-upgrade.nix#L112
`--no-build-output` is not recognized by `nixos-rebuild` in my configuration. If it does work elsewhere, it might be missing a check.

**Notify maintainers**

@Kloenk (last touched the module)

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.65, NixOS, 21.03.20200915.00bf6ab (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 3.0pre20200829_f156513`
 - channels(root): `"nixos-21.03pre246062.420f89ceb26"`
 - channels(astro): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module: nixos/autoUpgrade
```

**Comment by AstroSnail** [comment]
@edolstra (original creator of the module)

**Comment by FRidh** [comment]
I suppose you're using flakes? Should be fixed by https://github.com/NixOS/nixpkgs/pull/104870.

**Comment by AstroSnail** [comment]
It is fixed by #104870, thanks, closing.
